### ISSUE TITLE
[BUG] Matching locators by name is inaccurate due to escaped characters

### ISSUE THREAD

**Issue by lukaskl** [microsoft/playwright#20471]
### Context:
- Playwright Version: `1.30.0`
- Operating System: `Mac`
- Node.js version: `v18.12.1`
- Visual Studio Code version: Version: `1.74.3` (Universal), Commit: `97dec172d3256f8ca4bfb2143f3f76b503ca0534`
- Playwright for VSCode extension version: `1.0.6`
- Browser: `Chromium`

**Code Snippet**

```ts
import { test } from "@playwright/test";

test("What's Installed heading", async ({ page }) => {
  await page.goto("https://playwright.dev/docs/intro");

  await page.getByRole("heading", { name: "What's Installed#", exact: true }).hover();     // passes, VS Code extension doesn't work
  await page.getByRole("heading", { name: "What\'s Installed#", exact: true }).hover();    // passes, VS Code extension works
  await page.getByRole("heading", { name: "What\\'s Installed#", exact: true }).hover();   // passes, VS Code extension doesn't work
  await page.getByRole("heading", { name: "What\\\'s Installed#", exact: true }).hover();  // passes, VS Code extension doesn't work
  await page.getByRole("heading", { name: "What\\\\'s Installed#", exact: true }).hover(); // fails,  VS Code extension doesn't work
});
```

**Describe the bug**

I believe that the same cause leads to at least 2 issues:

- First, it is not accurate to allow to match locators by providing additional characters which do not exist on the browser (after all I'm passing a `string` not a `regex`). I believe that only the first `getByRole` here should pass and all the rest should fail:

  https://user-images.githubusercontent.com/4083977/215273296-5f5a77d4-1c5a-4782-9a65-027d9b03b5a7.mp4

- Second, VS Code extension picks up only the line with `"What\'s Installed#"` name (which I believe is ins't fully accurate) and doesn't show Locator tooltip when any other line (including `"What's Installed#"`) is selected/hovered:

  https://user-images.githubusercontent.com/4083977/215273727-45276f6f-edba-4c8b-8e23-f1e82b51e9cf.mp4

----- 

### Notes:

I personally believe that this is a minor issue, I've noticed accidentally it only when I was recording a video for [another issue](https://github.com/microsoft/playwright/issues/20470). I don't have a real use case which would be blocked by this. Nonetheless, I still thought it is worthy to report this issue.
